src/cli.rs

```diff
@@ -0,0 +1,501 @@
+        )
+        .arg(
+            Arg::with_name("derive_pk")
+                .long("derive_pk")
+                .help("Derive a 32-byte public key from a 612-byte signing key (stdin/file)")
+                .conflicts_with("generate_seed")
+                .conflicts_with("generate_sk")
+                .conflicts_with("derive_sk")
+                .takes_value(false),
+        )
+        .arg(
+            Arg::with_name("get_period")
+                .long("get_period")
+                .help("Get period from a 612-byte signing key (stdin/file)")
+                .conflicts_with("generate_seed")
+                .conflicts_with("generate_sk")
+                .conflicts_with("derive_sk")
+                .conflicts_with("derive_pk")
+                .takes_value(false),
+        )
+        .arg(
+            Arg::with_name("sign_msg")
+                .long("sign")
+                .help("Sign message (stdin) using a 612-byte signing key (file)")
+                .conflicts_with("generate_seed")
+                .conflicts_with("generate_sk")
+                .conflicts_with("derive_sk")
+                .conflicts_with("derive_pk")
+                .conflicts_with("get_period")
+                .takes_value(false),
+        )
+        .arg(
+            Arg::with_name("verify_sig")
+                .long("verify")
+                .help("Verified 64-byte signature using a message (stdin) and public key (file) for a given period")
+                .conflicts_with("generate_seed")
+                .conflicts_with("generate_sk")
+                .conflicts_with("derive_sk")
+                .conflicts_with("derive_pk")
+                .conflicts_with("get_period")
+                .conflicts_with("sign_msg")
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("update_sk")
+                .long("update_sk")
+                .help("Increment period for a 612 bytes signing key (stdin/file) which result in the updated signing key")
+                .conflicts_with("generate_seed")
+                .conflicts_with("generate_sk")
+                .conflicts_with("derive_sk")
+                .conflicts_with("derive_pk")
+                .conflicts_with("get_period")
+                .conflicts_with("sign_msg")
+                .conflicts_with("verify_sig")
+                .takes_value(false),
+        )
+        .arg(
+            Arg::with_name("file")
+                .value_name("FILE")
+                .help("Input file")
+                .multiple(false)
+                .default_value("-"),
+        )
+        .arg(
+            Arg::with_name("period")
+                .short("p")
+                .value_name("PERIOD")
+                .help("Number of updates a signing key experienced")
+                .multiple(false)
+                .takes_value(true)
+                .default_value("0"),
+        )
+        .get_matches();
+
+    Ok(if matches.is_present("generate_seed") {
```

Comment:
Having different subcommands structures would remove the need for this ugly if/else if tree